<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Germinate Example â€¢ QBMS</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Germinate Example">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QBMS</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/bms_example.html">BMS Example</a></li>
    <li><a class="dropdown-item" href="../articles/breedbase_example.html">BreedBase Example</a></li>
    <li><a class="dropdown-item" href="../articles/ebs_example.html">EBS Example</a></li>
    <li><a class="dropdown-item" href="../articles/germinate_example.html">Germinate Example</a></li>
    <li><a class="dropdown-item" href="../articles/gigwa_example.html">GIGWA Example</a></li>
    <li><a class="dropdown-item" href="../articles/hwsd2_example.html">FAO HWSD v2 Example</a></li>
    <li><a class="dropdown-item" href="../articles/terraclimate_example.html">TerraClimate Example</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/icarda/QBMS/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Germinate Example</h1>
                        <h4 data-toc-skip class="author">Khaled
Al-Shamaa</h4>
            
            <h4 data-toc-skip class="date">2025-07-31</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/icarda/QBMS/blob/HEAD/vignettes/germinate_example.Rmd" class="external-link"><code>vignettes/germinate_example.Rmd</code></a></small>
      <div class="d-none name"><code>germinate_example.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="qbms">QBMS<a class="anchor" aria-label="anchor" href="#qbms"></a>
</h2>
<p>This R package assists breeders in linking data systems with their
analytic pipelines, a crucial step in digitizing breeding processes. It
supports querying and retrieving phenotypic and genotypic data from
systems like <a href="https://ebs.excellenceinbreeding.org/" class="external-link">EBS</a>, <a href="https://bmspro.io/" class="external-link">BMS</a>, <a href="https://breedbase.org" class="external-link">BreedBase</a>, <a href="https://germinateplatform.github.io/get-germinate/" class="external-link">Germinate</a>
and <a href="https://github.com/SouthGreenPlatform/Gigwa2" class="external-link">GIGWA</a>
(using <a href="https://brapi.org/" class="external-link">BrAPI</a> calls). Extra helper
functions support environmental data sources, including <a href="https://www.climatologylab.org/terraclimate.html" class="external-link">TerraClimate</a>
and FAO <a href="https://gaez.fao.org/pages/hwsd" class="external-link">HWSDv2</a> soil
database.</p>
</div>
<div class="section level2">
<h2 id="germinate">Germinate<a class="anchor" aria-label="anchor" href="#germinate"></a>
</h2>
<p><a href="https://germinateplatform.github.io/get-germinate/" class="external-link">Germinate</a>
is an open source plant database infrastructure and application
programming platform on which complex data from genetic resource
collections can be stored, queried and visualized using common,
reusable, programming components. Germinate utilises modern web and
database standards to provide a common architecture and high performance
web-based user interface and analytics functionality across a wide
variety of data types including: passport, phenotypic, field trial,
pedigree, genetic, climatic, geographic location data as well as
user-submitted annotations. In addition, it provides links to an
expanding variety of external visualization and analysis tools such as
<a href="https://ics.hutton.ac.uk/helium/" class="external-link">Helium</a> for pedigree
visualization, <a href="https://ics.hutton.ac.uk/flapjack/" class="external-link">Flapjack</a>
for graphical genotyping and <a href="https://ics.hutton.ac.uk/curlywhirly/" class="external-link">CurlyWhirly</a> for 3D
coordinate data.</p>
</div>
<div class="section level2">
<h2 id="brapi">BrAPI<a class="anchor" aria-label="anchor" href="#brapi"></a>
</h2>
<p>The Breeding API (<a href="https://brapi.org/" class="external-link">BrAPI</a>) project is
an effort to enable interoperability among plant breeding databases.
BrAPI is a standardized RESTful web service API specification for
communicating plant breeding data. This community driven standard is
free to be used by anyone interested in plant breeding data
management.</p>
</div>
<div class="section level2">
<h2 id="connect-to-a-germinate-server">
<em>Connect to a Germinate server</em><a class="anchor" aria-label="anchor" href="#connect-to-a-germinate-server"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># load the QBMS library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://icarda.github.io/QBMS/" class="external-link">QBMS</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># configure your Germinate server connection (requires Germinate v4.9.0 or later)</span></span>
<span><span class="fu"><a href="../reference/set_qbms_config.html">set_qbms_config</a></span><span class="op">(</span>url <span class="op">=</span> <span class="st">"https://germinate.hutton.ac.uk/demo/#/home"</span>, </span>
<span>                engine <span class="op">=</span> <span class="st">"germinate"</span>, </span>
<span>                no_auth <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                page_size <span class="op">=</span> <span class="fl">9999</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># login using your account (interactive mode)</span></span>
<span><span class="co"># login()</span></span>
<span><span class="co"># or pass your username and password as parameters (batch mode)</span></span>
<span><span class="co"># login("username", "password")</span></span>
<span></span>
<span><span class="co"># list existing crops on the current server</span></span>
<span><span class="fu"><a href="../reference/list_crops.html">list_crops</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select a crop by name</span></span>
<span><span class="fu"><a href="../reference/set_crop.html">set_crop</a></span><span class="op">(</span><span class="st">"Cactuar"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># list all breeding programs in the selected crop</span></span>
<span><span class="fu"><a href="../reference/list_programs.html">list_programs</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select a breeding program by name</span></span>
<span><span class="fu"><a href="../reference/set_program.html">set_program</a></span><span class="op">(</span><span class="st">"Germinate"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># list all trials in the selected program</span></span>
<span><span class="fu"><a href="../reference/list_trials.html">list_trials</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select a specific trial by name</span></span>
<span><span class="fu"><a href="../reference/set_trial.html">set_trial</a></span><span class="op">(</span><span class="st">"GWAS data"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># list all studies (environments/locations) in the selected trial</span></span>
<span><span class="fu"><a href="../reference/list_studies.html">list_studies</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="work-with-phenotypic-data">
<em>Work with Phenotypic Data</em><a class="anchor" aria-label="anchor" href="#work-with-phenotypic-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select a specific study by name</span></span>
<span><span class="fu"><a href="../reference/set_study.html">set_study</a></span><span class="op">(</span><span class="st">"Sample Phenotype Data"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># retrieve general information and metadata for the selected study</span></span>
<span><span class="va">info</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_study_info.html">get_study_info</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># retrieve the germplasm list for the selected study</span></span>
<span><span class="va">geno</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_germplasm_list.html">get_germplasm_list</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># retrieve attributes for a specified germplasm</span></span>
<span><span class="va">attr</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_germplasm_attributes.html">get_germplasm_attributes</a></span><span class="op">(</span><span class="st">"CACTUAR-3"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># retrieve study data</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_study_data.html">get_study_data</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get observation variable ontology in the selected study/trial</span></span>
<span><span class="va">ontology</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_trial_obs_ontology.html">get_trial_obs_ontology</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="work-with-genotypic-data">
<em>Work with Genotypic Data</em><a class="anchor" aria-label="anchor" href="#work-with-genotypic-data"></a>
</h2>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select a specific study by name</span></span>
<span><span class="fu"><a href="../reference/set_study.html">set_study</a></span><span class="op">(</span><span class="st">"Sample Genotype Data Subset 2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># list all variant sets in the selected study</span></span>
<span><span class="fu"><a href="../reference/list_variantsets.html">list_variantsets</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select a specific active variant set by name</span></span>
<span><span class="fu"><a href="../reference/set_variantset.html">set_variantset</a></span><span class="op">(</span><span class="st">"Sample Genotype Data Subset 2"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># retrieve the selected variant set marker matrix</span></span>
<span><span class="va">snps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_variantset.html">get_variantset</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get marker position information</span></span>
<span><span class="va">map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_marker_map.html">get_marker_map</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="work-with-pedigree-data">
<em>Work with Pedigree Data</em><a class="anchor" aria-label="anchor" href="#work-with-pedigree-data"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># select a specific study by name</span></span>
<span><span class="fu"><a href="../reference/set_trial.html">set_trial</a></span><span class="op">(</span><span class="st">"Default pedigree experiment"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get the pedigree table for the selected trial</span></span>
<span><span class="va">ped</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_trial_pedigree.html">get_trial_pedigree</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Khaled Al-Shamaa.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
